crates/bevy_asset/src/io/mod.rs

```diff
@@ -39,6 +39,7 @@ pub trait AssetIo: Downcast + Send + Sync + 'static {
         &self,
         path: &Path,
     ) -> Result<Box<dyn Iterator<Item = PathBuf>>, AssetIoError>;
+    fn is_file(&self, path: &Path) -> bool;
     fn is_directory(&self, path: &Path) -> bool;
```

Comment:
This may be out of scope for this PR, however I'd be wary of adding on functions like this to the `AssetIo` trait. This could keep expanding and making the class larger and larger.

An alternative to consider it have a `metadata` function which
```rust
fn metadata(&self, path: &Path) -> Result<std::fs::Metadata>;
```

This way the metadata function coalesces the `is_directory`/`is_file` methods.

Reply:
Good idea. But `std::fs::Metadata` cannot be used with archives like in [bevy_assetio_zip](https://github.com/katharostech/bevy_assetio_zip). Am i right?

Reply:
Hmm, I'm not very familiar with `bevy_assetio_zip`, though after taking a brief look at the implementation, it uses a `Box<dyn AssetIo>` under the hood which would allow you to call `metadata()` on it.
So I'm not sure how using `std::fs::Metadata` would affect the crate, but again I could be wrong haha 🙃

Reply:
Maybe it was a bad example. 😄 My projects for example uses multiple archives and simulates a filesystem on top of it. So I cannot use  `std::fs::Metadata`.

Reply:
AssetIo is a "virtual filesystem", but it isn't a real filesystem as defined by `std::fs`. I'm cool with a Metadata struct that holds "virtual filesystem metadata" for a given path, but we should define that ourselves and scope it specifically to things that any "virtual filesystem" could provide. Probably just "is_file" and "is_directory" for now.